reana_db/database.py

```diff
@@ -23,10 +24,19 @@
 Base.query = Session.query_property()
 
 
+def _before_create(target, connection, **kwargs):
+    """Create schemas as SQLAlchemy doesn't do it and set 'public' schema as default."""
+    CreateSchema("reana").execute(connection)
+    # Otherwise Invenio tables end up in 'reana' schema.
+    sql = "ALTER ROLE {} SET search_path TO public".format(DB_USERNAME)
```

Comment:
I couldn't find a better/programmatic solution to run this statement. I'm afraid this can be vulnerable to SQL injection attacks.. although I've tried many `DB_USERNAME` and I didn't succeed e.g `"reana'; SELECT hacked"` and so on.
Do you have any ideas to improve this?

Reply:
I know this would be making assumptions, but since we are the ones in charge of the code that instantiates the DBs, what if we invert the order and we move the [Invenio DB creation](https://github.com/reanahub/reana-server/blob/486c96d4968243fff1d3faaba21dc219f2ee20af/scripts/setup#L18) before the REANA DB creation? Wouldn't that solve it?

Reply:
It works but then an empty revision contains this:

```python
ef upgrade():
    # ### commands auto generated by Alembic - please adjust! ###
    op.create_foreign_key(None, 'audit_log', 'user_', ['user_id'], ['id_'], source_schema='reana', referent_schema='reana')
    op.create_foreign_key(None, 'job_cache', 'job', ['job_id'], ['id_'], source_schema='reana', referent_schema='reana')
    op.create_foreign_key(None, 'user_token', 'user_', ['user_id'], ['id_'], source_schema='reana', referent_schema='reana')
    op.create_foreign_key(None, 'workflow', 'user_', ['owner_id'], ['id_'], source_schema='reana', referent_schema='reana')
    # ### end Alembic commands ###


def downgrade():
    # ### commands auto generated by Alembic - please adjust! ###
    op.drop_constraint(None, 'workflow', schema='reana', type_='foreignkey')
    op.drop_constraint(None, 'user_token', schema='reana', type_='foreignkey')
    op.drop_constraint(None, 'job_cache', schema='reana', type_='foreignkey')
    op.drop_constraint(None, 'audit_log', schema='reana', type_='foreignkey')
    # ### end Alembic commands ###
```

Reply:
Check [`quote_ident`](https://www.psycopg.org/docs/extensions.html#psycopg2.extensions.quote_ident) to escape `DB_USERNAME`.

Reply:
Changing schema name solves this problem, we don't need `ALTER ROLE` anymore.